syntax = "proto3";

package bookstore;

//import "src/app/grpc/protos/books.proto";
import "books.proto";

service PurchaseService {
  rpc ListPurchases (Empty) returns (ListPurchasesResponse);
  rpc CreatePurchase (CreatePurchaseRequest) returns (CreatePurchaseResponse);
}

message Purchase {
  string book_title = 1;
  int32 quantity = 2;
  double total_price = 3;
  string date = 4;
}

message ListPurchasesResponse {
  repeated Purchase purchases = 1;
}

message CreatePurchaseRequest {
  int32 book_id = 1;
  int32 quantity = 2;
}

message CreatePurchaseResponse {
  bool success = 1;
}
